AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift

```diff
@@ -150,13 +146,61 @@ class DataStoreEndToEndTests: SyncEngineIntegrationTestBase {
                     createReceived.fulfill()
                     return
                 }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        Amplify.DataStore.save(newPost) { _ in }
+        await waitForExpectations(timeout: networkTimeout)
+
+        let updateReceived = expectation(description: "Update notification received")
+        hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+                guard let mutationEvent = payload.data as? MutationEvent
+                    else {
+                        XCTFail("Can't cast payload as mutation event")
+                        return
+                }
+                // This check is to protect against stray events being processed after the test has completed,
+                // and it shouldn't be construed as a pattern necessary for production applications.
+                guard let post = try? mutationEvent.decodeModel() as? Post, post.id == newPost.id else {
+                    return
+                }
 
                 if mutationEvent.mutationType == GraphQLMutationType.update.rawValue {
                     XCTAssertEqual(post.content, updatedPost.content)
                     XCTAssertEqual(mutationEvent.version, 2)
                     updateReceived.fulfill()
                     return
                 }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
```

Comment:
Recommend we update `HubListenerTestUtilities.waitForListener(...)` to be async and then use `try await Task.sleep(seconds:)` instead of `Thread.sleep(forTimeInterval:)` in that function.  This will make it create a Swift Concurrency suspension instead of blocking the Thread.  That can be a separate PR though.

Reply:
see [this PR](https://github.com/aws-amplify/amplify-ios/pull/2084) for a convenience wrapper

Reply:
Thanks! I checked the usage of this and it spans a couple of places, so will cut a new PR for this

Reply:
https://github.com/aws-amplify/amplify-ios/pull/2087